### ISSUE TITLE
calculate_fisher doesn't sync gradients for DDP model.

### ISSUE THREAD

**Issue by Cecilwang** [kazukiosawa/asdfghjkl#27]
1. We want ASDL to support distributed training with [DistributedDataParallel](https://pytorch.org/docs/stable/nn.html#torch.nn.parallel.DistributedDataParallel) but it cannot process DDP model now.
```
model = resetnet50();
kfac = asdl.KFAC(model, 'fisher_emp'); # must create KFAC before wrapping
model = DistributedDataParallel(model, device_ids=[args.gpu])
```
2. DDP will register sync hook for every parameter to reduce all gradients when calling `loss.backward()`. But gradients are not syncing after `kfac.accumulate_curvature`.
```
criterion(model(inputs), targets).backward(); # gradients will be synced.
kfac.accumulate_curvature(inputs, targets, calc_emp_loss_grad=True); # gradients will not be synced
```
We don't know why it happens even if `backward` will be called in `accumulate_curvature`.
